batnode.js

```diff
@@ -250,25 +251,59 @@ class BatNode {
       this.getHostNode(currentCopy, afterHostNodeIsFound)
     }
   }
+  combineShardsAfterWaitTime(waitTime, fileName, distinctShards) {
+    return new Promise(resolve => {
+      setTimeout(() => resolve(fileUtils.assembleShards(fileName, distinctShards)), waitTime);
+    });
+  }
+
+  async asyncCallAssembleShards(waitTime, fileName, distinctShards) {
+    try {
+      console.log("waiting time in ms: ", waitTime);
+      const result = await this.combineShardsAfterWaitTime(waitTime, fileName, distinctShards);
+      return result;
+    } catch (e) {
+      console.error(e);
+    }
+  }
 
-  issueRetrieveShardRequest(shardId, hostBatNode, options, finishCallback){
+  issueRetrieveShardRequest(shardId, hostBatNode, manifest, options, finishCallback){
```

Comment:
Is this param `manifest` the loaded file in JSON format. What do you think about calling it `manifestJson`?

Reply:
Good idea, will change it to manifestJson

Reply:
Rename the params names in new commit: 5e134a6a7d0cdbb6fa4b718c81515bb1457814d6